File: 1768-merge-strings-alternately/1768-merge-strings-alternately.js
/**
 * @param {string} word1
 * @param {string} word2
 * @return {string}
 */
var mergeAlternately = function(word1, word2) {
    let dist = Math.max(word1.length, word2.length)

    let out = ""
    for (let i = 0; i < dist; i++) {
        if (i < word1.length) {
            out += word1.charAt(i);
        }
        if (i < word2.length) {
            out += word2.charAt(i);
        }
    }

    return out
};

File: 1768-merge-strings-alternately/1768-merge-strings-alternately.py
class Solution:
    def mergeAlternately(self, word1: str, word2: str) -> str:
        dist = max(len(word1), len(word2))

        out = ""
        for i in range(dist):
            if i < len(word1):
                out += word1[i]
            if i < len(word2):
                out += word2[i]
        
        return out